src/Nethermind/Nethermind.Network/TrustedNodes/TrustedNodesManager.cs

```diff
@@ -0,0 +1,172 @@
+            List<NetworkNode> networkNodes = new();
+            foreach (string line in lines)
+            {
+                try
+                {
+                    networkNodes.Add(new NetworkNode(line));
+                }
+                catch (Exception ex) when (ex is ArgumentException or SocketException)
+                {
+                    if (_logger.IsError)
+                    {
+                        _logger.Error($"Failed to parse '{line}' as a trusted node.", ex);
+                    }
+                }
+            }
+
+            _nodes = new ConcurrentDictionary<PublicKey, NetworkNode>(
+                networkNodes.ToDictionary(n => n.NodeId, n => n));
+        }
+
+        // ---- INodeSource requirement: IAsyncEnumerable<Node> ----
+        // C# requires 'async' for IAsyncEnumerable yield. We'll add a small 'await Task.Yield()'
+        // to avoid the warning about "no awaits".
+        public async IAsyncEnumerable<Node> DiscoverNodes([EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            // At least one 'await', so no compiler warnings
+            await Task.Yield();
+
+            foreach (NetworkNode netNode in _nodes.Values)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                yield return new Node(netNode) { IsTrusted = true };
+            }
+        }
+
+        public async Task<bool> AddAsync(Enode enode, bool updateFile = true)
+        {
+            NetworkNode networkNode = new(enode.ToString());
+            if (!_nodes.TryAdd(networkNode.NodeId, networkNode))
+            {
+                if (_logger.IsInfo)
+                {
+                    _logger.Info($"Trusted node was already added: {enode}");
+                }
+                return false;
+            }
+
+            if (_logger.IsInfo)
+            {
+                _logger.Info($"Trusted node added: {enode}");
+            }
+
+            if (updateFile)
+            {
+                await SaveFileAsync();
+            }
+            return true;
+        }
+
+        public async Task<bool> RemoveAsync(Enode enode, bool updateFile = true)
+        {
+            NetworkNode networkNode = new(enode.ToString());
+            if (!_nodes.TryRemove(networkNode.NodeId, out _))
+            {
+                if (_logger.IsInfo)
+                {
+                    _logger.Info($"Trusted node was not found: {enode}");
+                }
+                return false;
+            }
+
+            if (_logger.IsInfo)
+            {
+                _logger.Info($"Trusted node was removed: {enode}");
+            }
+
+            if (updateFile)
+            {
+                await SaveFileAsync();
+            }
+            return true;
+        }
+
+        public bool IsTrusted(Enode enode)
+        {
+            NetworkNode node = new(enode.ToString());
+            return _nodes.ContainsKey(node.NodeId);
+        }
```

Comment:
You convert `Node` `ToString`, to convert it to `Enode`, to pass here, to convert it back `ToString`, to create `NetworkNode`, to get `NodeId`, while you had `node.Id` right there in `PeerPool.CreateNew` to begin with as `PublicKey`.

Maybe just make it `public bool IsTrusted(PublicKey key)` ?

Reply:
Or maybe we should also check other parts of Enode like Host and Port and not only PublicKey?